apps/website/src/components/HomepageBanner.astro

```diff
@@ -0,0 +1,70 @@
+---
+import RightArrowIcon from '~/icons/RightArrowIcon.astro';
+
+type Props = {
+  href: string;
+  [key: string]: unknown;
+};
+
+const { href, ...props } = Astro.props;
+---
+
+<aside {...props}>
+  <a href={href}>
+    <slot />
+  </a>
+
+  <RightArrowIcon class='arrow' lineProps={{ class: 'line' }} />
+</aside>
+
+<style lang='scss'>
+  @layer components {
+    aside {
+      border: 1px solid var(--color-line-2);
+      background-color: var(--color-background-1);
+      border-radius: 9e9px;
+      display: inline-flex;
+      align-items: center;
+      color: var(--color-text);
+      padding-inline: 16px 8px;
+      padding-block: 4px;
+      gap: 4px;
+      cursor: pointer;
+      transition: border-color 0.2s ease-in-out;
+
+      :global(.line) {
+        opacity: 0;
+        transition: opacity 0.2s ease-in-out;
+      }
+    }
+
+    a {
+      all: unset;
+    }
```

Comment:
Is having global styles with a generic name like `line` alright?

Reply:
it's not truly global, it's nested under `aside` which is scoped.

this is because astro cannot scope indirectly passed classes (in this case using `lineProps`)